### ISSUE TITLE
open Modal Dialog Navigation from router-outlet component cause problem.

### ISSUE THREAD

**Issue by liuwei108** [NativeScript/nativescript-angular#1467]
### If there is no issue for your problem, tell us about it
If Modal Dialog Navigation was opened from a component inside router-outlet, and router.back() was called during navigation in modal, the second time to open this modal will show a blank page and sometimes below error generated:

> JS: TypeError: Cannot read property 'key' of undefined
> JS:     at NSRouteReuseStrategy.shouldAttach (file:///data/data/org.nativescript.Nsproject/files/app/tns_modules/nativescript-angular/router/ns-route-reuse-strategy.js:91:50)
> JS:     at ActivateRoutes.activateRoutes (file:///data/data/org.nativescript.Nsproject/files/app/tns_modules/@angular/router/bundles/router.umd.js:4026:49)
> JS:     at file:///data/data/org.nativescript.Nsproject/files/app/tns_modules/@angular/router/bundles/router.umd.js:3999:23
> JS:     at Array.forEach (<anonymous>)
> JS:     at ActivateRoutes.activateChildRoutes (file:///data/data/org.nativescript.Nsproject/files/app/tns_modules/@angular/router/bundles/router.umd.js:3998:33)
> JS:     at ActivateRoutes.activateRoutes (file:///data/data/org.nativescript.Nsproject/files/app/tns_modules/@angular/router/bundles/router.umd.js:4015:26)
> JS:     at file:///data/data/org.nativescript.Nsproject/files/app/tns_modules/@angular/router/bun...
> 

### Which platform(s) does your issue occur on?
- iOS/Android/Both 

### Please, provide the following version numbers that your issue occurs with:

{
  "description": "NativeScript Application",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "org.nativescript.Nsproject",
    "tns-android": {
      "version": "4.1.3"
    },
    "tns-ios": {
      "version": "4.1.0"
    }
  },
  "dependencies": {
    "@angular/animations": "~6.0.6",
    "@angular/common": "~6.0.6",
    "@angular/compiler": "~6.0.6",
    "@angular/core": "~6.0.6",
    "@angular/forms": "~6.0.6",
    "@angular/http": "~6.0.6",
    "@angular/platform-browser": "~6.0.6",
    "@angular/platform-browser-dynamic": "~6.0.6",
    "@angular/router": "~6.0.6",
    "@nativescript/schematics": "^0.1.5",
    "@ngrx/effects": "^6.0.1",
    "@ngrx/store": "^6.0.1",
    "nativescript-angular": "^6.0.6",
    "nativescript-theme-core": "~1.0.4",
    "nativescript-ui-sidedrawer": "^4.1.1",
    "reflect-metadata": "~0.1.8",
    "rxjs": "~6.0.0 || >=6.1.0",
    "tns-core-modules": "^4.1.1",
    "zone.js": "^0.8.26"
  },
  "devDependencies": {
    "@angular/cli": "^6.0.8",
    "@ngrx/schematics": "^6.0.1",
    "babel-traverse": "6.26.0",
    "babel-types": "6.26.0",
    "babylon": "6.18.0",
    "lazy": "1.0.11",
    "nativescript-dev-typescript": "~0.7.0",
    "typescript": "~2.7.2"
  }
}

**Comment by tsonevn** [comment]
HI @liuwei108,
I tested the case on my side, however, was unable to recreate the similar problem. For your convenience, I am attaching a sample project. Please review it and make the needed changes, which will allow us to recreate the issue.
[Archive.zip](https://github.com/NativeScript/nativescript-angular/files/2248225/Archive.zip)

**Comment by liuwei108** [comment]
Hello Tsonevn,

Sorry I forgot to mention another factor: `the path contains named router outlet.`
Here please test this [demo](https://play.nativescript.org/?template=play-ng&id=H6aK0A&v=8) in playgroup

**Comment by tsonevn** [comment]
Hi @liuwei108,
Thank you for the sample project. I tested the app, however, I am still unable to recreate the issue. For your convenience, I am attaching a video and the log from the console. 
[test.zip](https://github.com/NativeScript/nativescript-angular/files/2252386/test.zip)


```
09:18:01 - Compilation complete. Watching for file changes.



CONSOLE LOG file:///app/modal/modal.component.js:12:20: Modal init
CONSOLE LOG file:///app/landing/landing.component.js:21:28: return value
CONSOLE LOG file:///app/modal/modal.component.js:12:20: Modal init
CONSOLE LOG file:///app/landing/landing.component.js:21:28: success
CONSOLE LOG file:///app/modal/modal.component.js:12:20: Modal init
CONSOLE LOG file:///app/landing/landing.component.js:21:28: return value
JS: Modal init
JS: return value
JS: Modal init
JS: success
```
I tested the project on a simulator with iOS 11.4 and  Android API level 24.

**Comment by liuwei108** [comment]
Hello @tsonevn ,

Have you navigate from login to signup page then click back to login and close the modal?

1. click `open Modals` button to open modal, login component appears
2. click `Sign Up` button to navigate to Signup component.
3. click `Back` button to back to login component, here `router.back()` invoked, this is the key to reproduce this problem
4. close `CLOSE MODAL`, then click `open Modals`, you should see blank page now.

**Comment by tsonevn** [comment]
Hi @liuwei108,
I discussed the issue with our dev team and we concluded that this issue is related to the feature request logged [here](https://github.com/NativeScript/nativescript-angular/issues/1351). You can keep track on it for further info.

**Comment by liuwei108** [comment]
Hello @tsonevn 
Thanks, Just want to reminder that my issue only happens in the condition of opening Modal Dialog Navigation, I can't see modal was mentioned in #1351, would be appreciated if you can double check if this issue can be resolved by implementing feature #1351.
